### ISSUE TITLE
Can't create schema due to cyclic fk references

### ISSUE THREAD

**Issue by danieldaeschle** [tortoise/tortoise-orm#154]
I have a User model.
I want a one to one relation between two users.
```python
spouse = fields.ForeignKeyField('models.User', on_delete=fields.SET_NULL, null=True)
```

But i get this error:
```
tortoise.exceptions.ConfigurationError: Can't create schema due to cyclic fk references
```

Any help?

**Comment by grigi** [comment]
Ah, we need to formal add support for "recursive" foreign keys? Right now the cyclic detector blocks it regardless if it works or not.

**Comment by harshanarayana** [comment]
@grigi This seems like a small change under `get_create_schema_sql` method to enable this feature. 

```python
next_table_for_create = next(
    t for t in tables_to_create if t["references"].issubset(created_tables)
    or t["references"].issubset({t["table"]})
)
```

Am I getting it wrong? Also, was there a reason this was disabled initially?

**Comment by grigi** [comment]
I think it was just a sanity protection? Cyclic references means we don't know which table to create.
If it is just that, then should be safe.

Recursive functions tend to have an implicit race condition where they are not yet created but already referenced.
As long as there is a simple test case to validate it works as expected?

**Comment by grigi** [comment]
@harshanarayana Are you working on this? It is unclear?

**Comment by harshanarayana** [comment]
@grigi  I will take a look at this later today. I need to think of a clean way to add a few tests cases to ensure race condition checks. I think I will be able to have a PR up for this in a day or two. Would that work?

**Comment by grigi** [comment]
That would be great!
A test like this one for error cases: https://github.com/tortoise/tortoise-orm/blob/master/tortoise/tests/test_generate_schema.py#L89 (it uses a different model: https://github.com/tortoise/tortoise-orm/blob/master/tortoise/tests/models_cyclic.py)
And cases we want to be successful you can do in the generic `models.py`.

**Comment by harshanarayana** [comment]
@grigi I am sorry I haven't been able to get to this in last two days. I am going to look at this over the weekend and get it done. Sorry for the delay.

**Comment by grigi** [comment]
Thanks for your investigative work @harshanarayana 
I released v0.12.7 that should fix this issue @danieldaeschle 
Please test if it resolves your issue

**Comment by FIRDOUS-BHAT** [comment]
Hello there,
  I'm getting the same error after I removed the foreign key from the schema
